File: json.cs
using System;
using System.Text;
using System.IO;
using System.Linq;

public class Program
{
  private static readonly string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";

  public static string RanCode(int leng)
  {
  return new string(Enumerable.Repeat(chars, leng).Select(s => s[random.Next(s.Length)]).ToArray());
  }

  public static void FileValue(string name, string exp, dynamic data)
  {
  using (StreamWriter file = new StreamWriter($"{nam e}.{exp}"))
  {
  file.Write(JsonConvert.SerializeObject(data));
  }
  }

  public static void Main()
  {
  Console.Write("Введите имя - ");
  string username = Console.ReadLine();
  int passwordLength = random.Next(10, 20);
  string password = RanCode(passwordLength);
  dynamic data = new
  {
  username,
  password  iflogin = true
  };

  FileValue("File", "json", data);
  }
}

File: json.pas
program CodeConverter;

var
  username: string;
  password: string;
  datafile: string;

function RanCode(leng: integer): string;
begin
  var
  charSet: set of char;
  choice: char;
  begin
  charSet := 'abcdefghijklmnopqrstuvwxyz0123456789';
  for i := 1 to leng do
  choice := Random(charSet);
  Result := choice;
  end;
end;

procedure FileValue(name, extension, data: string);
var
  file: text;
begin
  AssignFile(file, name+, true); // Opens and creates the file if it doesn exist
  Writeln(file, jsontoString(data));
  CloseFile(file);
end;

function jsontoString(data: string): string;
var
  i: integer = 1;
begin := '{"username": "' + username + '",';
  Result := Result + ' "password":"' + password + '",  Result := Result + ' "iflogin?":true}';
end;

begin
  writeln('Введите имя:');
  Readln(username);
  write('Password: ');
  Write(password :=Code(random(10) + 1));
  datafile := '{"username":"'+ username +'",';
  datafile := datafile + ' "password":"' + password + '",';
  datafile := datafile + '"iflogin?":true}';
  FileValue('File', '.json', datafile);
end.